### PR TITLE
implement js-routing for commerce

### PR DESCRIPTION
**Pull request by dpfaffenbauer**
Not sure whether you want this in 6.6 or dev-master

### CODE REVIEW

**Comment by dvesh3** [comment]
@dpfaffenbauer completely resolves https://github.com/pimcore/pimcore/issues/6481?

**Comment by dpfaffenbauer** [comment]
>  - Replace ecommerce routes with Route Generation using FOSJS

Yes

> - Replace all static route uses in PHP and Templates

No, cause it already used route-names

>  - Add route names for @Route annotation (auto-generated names for BC reasons)

Yes

**Comment by brusch** [comment]
@dpfaffenbauer thanks a lot! 
`master` would be nice, can you please rebase? 
Thanks again!

**Comment by dpfaffenbauer** [comment]
@brusch done

**Comment by brusch** [comment]
@dpfaffenbauer Grazie Mille! 👌😊

File: bundles/EcommerceFrameworkBundle/Resources/public/js/bundle.js [hunk 1]
```diff
@@ -132,7 +132,7 @@ pimcore.bundle.EcommerceFramework.bundle = Class.create(pimcore.plugin.admin, {
         var that = this;
 
         Ext.Ajax.request({
-            url: '/admin/ecommerceframework/reports/piwik/reports',
+            url: Routing.generate('pimcore_ecommerceframework_reports_piwik_reports'),
```
Comment (dvesh3) [inline bundles/EcommerceFrameworkBundle/Resources/public/js/bundle.js:135]: Bundle loading failed with route `pimcore_ecommerceframework_reports_piwik_reports` does not exist error
image

**Comment by dvesh3** [comment]
@dpfaffenbauer please see https://github.com/pimcore/pimcore/pull/6862#discussion_r464895249

File: bundles/EcommerceFrameworkBundle/Resources/public/js/bundle.js (hunk 1 above)
Comment (dpfaffenbauer) [reply bundles/EcommerceFrameworkBundle/Resources/public/js/bundle.js:135]: fixed

File: bundles/EcommerceFrameworkBundle/Controller/PricingController.php [hunk 2]
```diff
@@ -194,7 +194,7 @@ public function deleteAction(Request $request)
     /**
      * save rule config
      *
-     * @Route("/save", methods={"PUT"})
+     * @Route("/save", name="pimcore_ecommerceframework_pricing_save", methods={"PUT"})
```
Comment (dvesh3) [inline bundles/EcommerceFrameworkBundle/Controller/PricingController.php:197]: please update js: https://github.com/pimcore/pimcore/blob/53e8d494faea79fc17c7e43071f0bbb59a4b7bfe/bundles/EcommerceFrameworkBundle/Resources/public/js/pricing/config/item.js#L416

File: bundles/EcommerceFrameworkBundle/Controller/PricingController.php [hunk 3]
```diff
@@ -46,7 +46,7 @@ public function onKernelController(FilterControllerEvent $event)
     }
 
     /**
-     * @Route("/list", methods={"GET"})
+     * @Route("/list", name="pimcore_ecommerceframework_pricing_list", methods={"GET"})
```
Comment (dvesh3) [inline bundles/EcommerceFrameworkBundle/Controller/PricingController.php:49]: please update js:
https://github.com/pimcore/pimcore/blob/c62924ca5c3d0c482a71097cb0231aff143208d6/bundles/EcommerceFrameworkBundle/Resources/public/js/pricing/config/panel.js#L160
and others too:
https://github.com/pimcore/pimcore/blob/c62924ca5c3d0c482a71097cb0231aff143208d6/bundles/EcommerceFrameworkBundle/Resources/public/js/pricing/config/panel.js#L50
https://github.com/pimcore/pimcore/blob/c62924ca5c3d0c482a71097cb0231aff143208d6/bundles/EcommerceFrameworkBundle/Resources/public/js/pricing/config/panel.js#L142
https://github.com/pimcore/pimcore/blob/c62924ca5c3d0c482a71097cb0231aff143208d6/bundles/EcommerceFrameworkBundle/Resources/public/js/pricing/config/panel.js#L249
https://github.com/pimcore/pimcore/blob/c62924ca5c3d0c482a71097cb0231aff143208d6/bundles/EcommerceFrameworkBundle/Resources/public/js/pricing/config/panel.js#L287
https://github.com/pimcore/pimcore/blob/c62924ca5c3d0c482a71097cb0231aff143208d6/bundles/EcommerceFrameworkBundle/Resources/public/js/pricing/config/panel.js#L316

File: bundles/EcommerceFrameworkBundle/Controller/PricingController.php (hunk 2 above)
Comment (dpfaffenbauer) [reply bundles/EcommerceFrameworkBundle/Controller/PricingController.php:197]: done

File: bundles/EcommerceFrameworkBundle/Controller/PricingController.php (hunk 3 above)
Comment (dpfaffenbauer) [reply bundles/EcommerceFrameworkBundle/Controller/PricingController.php:49]: done

File: bundles/EcommerceFrameworkBundle/Controller/IndexController.php [hunk 4]
```diff
@@ -15,6 +15,7 @@
 namespace Pimcore\Bundle\EcommerceFrameworkBundle\Controller;
 
 use Pimcore\Bundle\AdminBundle\Controller\AdminController;
+use Pimcore\Bundle\AdminBundle\HttpFoundation\JsonResponse;
```
Comment (dvesh3) [inline bundles/EcommerceFrameworkBundle/Controller/IndexController.php:18]: unused

**Comment by dpfaffenbauer** [comment]
done

**Comment by dvesh3** [comment]
thanks a lot! 👏

**Pull request merged**